### ISSUE TITLE
Old service versions should not be removed in case of deployment failures

### ISSUE THREAD

**Issue by cristim** [serverless/serverless#2458]
# This is a Bug Report

## Description

* What went wrong?
 - After multiple failures to run the CloudFormation code, the stack gets into an unrecoverable state and needs to be deleted and re-created, because all the ZIP files eventually get deleted from S3 and the stack fails to rollback to the previous stack version.
* What did you expect should have happened?
 - The stack should be able to rollback cleanly, by keeping the ZIP files in the bucket in case of stack update failures.
* What was the config you used?
 - I can't paste it here, but it was an initially working stack where I added a resource with incorrect CloudFormation code(anything should do, in my case it was an AWS::CloudFront::Distribution which I had hard time getting right). After performing a number of updates with incorrect CloudFormation code the stack ended up in a broken state and needed to be deleted and re-created.
* What stacktrace or error message from your provider did you see?
```
Serverless: Wrote .env file to /home/magherus/work/.env
Serverless: Packaging service...
Serverless: Deleted .env file from /home/magherus/work/.env
Serverless: Removing old service versions...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading service .zip file to S3...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
CloudFormation - UPDATE_IN_PROGRESS - AWS::CloudFormation::Stack - my-stack
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction2
CloudFormation - CREATE_IN_PROGRESS - AWS::CloudFront::Distribution - CloudFrontDistribution
CloudFormation - CREATE_FAILED - AWS::CloudFront::Distribution - CloudFrontDistribution
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction3
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction2
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction3
CloudFormation - UPDATE_ROLLBACK_IN_PROGRESS - AWS::CloudFormation::Stack - my-stack
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction2
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction2
CloudFormation - UPDATE_IN_PROGRESS - AWS::Lambda::Function - MyLambdaFunction3
CloudFormation - UPDATE_FAILED - AWS::Lambda::Function - MyLambdaFunction3
CloudFormation - UPDATE_ROLLBACK_FAILED - AWS::CloudFormation::Stack - my-stack
Serverless: Deployment failed!
 
  Serverless Error ---------------------------------------
 
     An error occurred while provisioning your stack: CloudFrontDistribution
     - The parameter originPath contains one or more parameters
     that are not valid..
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
```

## Additional Data

* ***Serverless Framework Version you're using***: 1.0.0-rc2
* ***Operating System***: Ubuntu 16.04
* ***Provider Error messages***: see above

**Comment by WooDzu** [comment]
I think this one is fixed in 1.0.2

**Comment by cristim** [comment]
Thanks @WooDzu, I am attempting to update to 1.0.2 but unfortunately the update fails, apparently the 1.0.2 minor version breaks compatibility with stacks which used to work in 1.0.0-rc2. I will raise this on another issue.

**Comment by flomotlik** [comment]
There is a PR for this: https://github.com/serverless/serverless/pull/2348 Should be merged soonish so it works then. As this is already being addressed in a PR and is a duplicate of #2011 I'm closing here.
